usv_gazebo_plugins/src/usv_gazebo_dynamics_plugin.cc

```diff
@@ -328,7 +328,9 @@ void UsvDynamicsPlugin::Update()
       double simTime = kTimeNow.Double();
       // double depth = WavefieldSampler::ComputeDepthDirectly(
       //  *waveParams, X, simTime);
-      double depth = WavefieldSampler::ComputeDepthSimply(
+      double depth = 0.0;
+      if (waveParams)
+        depth = WavefieldSampler::ComputeDepthSimply(
         *waveParams, X, simTime);
```

Comment:
Actually I'm gonna be nitpicky here and suggest style changes
```suggestion
      if (waveParams)
      {
        depth = WavefieldSampler::ComputeDepthSimply(*waveParams, X, simTime);
      }
```